### PR TITLE
Create module for SB subscription SQL-filter

### PR DESCRIPTION
**Pull request by kerbou**
Ref: https://app.zenhub.com/workspaces/mighty-ducks---the-outlaws-6193fe815d79fc0011e741b1/issues/energinet-datahub/the-outlaws/505

### CODE REVIEW

File: azure/service-bus-topic-subscription/main.tf [hunk 1]
```diff
@@ -18,4 +18,12 @@ resource "azurerm_servicebus_subscription" "this" {
 
   max_delivery_count  = try(var.max_delivery_count, null)
   forward_to          = try(var.forward_to, null)
+}
+
+
+resource "azurerm_servicebus_subscription_rule" "sqlfilter" {
```
Comment (djorgensendk) [inline azure/service-bus-topic-subscription/main.tf:24]: Adding a filter is optional (according to the documentation of the sqlfilter_name variable.
To make sure no resource is created please add an "if-statement" using count. See example in storage-account/main.tf line 89

File: azure/service-bus-topic-subscription/README.md [hunk 2]
```diff
@@ -24,18 +25,29 @@ See [variables.tf](./variables.tf)
 ## Usage
 
 
-
+### Subscription without filters
 ```ruby
-
 module "sbtsub_projectname_subscriptionname" {
```
Comment (Renetnielsen) [inline azure/service-bus-topic-subscription/README.md:30]: To adhere with how we've created the other examples i would just put in `sbtsub_example`

See for instance
https://github.com/Energinet-DataHub/geh-terraform-modules/blob/main/azure/app-service/README.md
https://github.com/Energinet-DataHub/geh-terraform-modules/blob/main/azure/mssql-database/README.md

etc

File: azure/service-bus-topic-subscription/README.md [hunk 3]
```diff
@@ -24,18 +25,29 @@ See [variables.tf](./variables.tf)
...
   name                = "examplesubscription"
   project_name        = var.project_name  
   topic_id            = "/subscriptions/<subscription id>/resourceGroups/<resourcegroup>/providers/Microsoft.ServiceBus/namespaces/<servicebus namespace>/topics/<topicname>"
   max_delivery_count  = 1
 }
 ```
 
+### Subscription with SQL filter applied
+```ruby
+module "sbtsub_projectname_subscriptionname_withsqlfilter" {
```
Comment (Renetnielsen) [inline azure/service-bus-topic-subscription/README.md:41]: To adhere with how we've created the other examples i would just put in sbtsub_example

See for instance
https://github.com/Energinet-DataHub/geh-terraform-modules/blob/main/azure/app-service/README.md
https://github.com/Energinet-DataHub/geh-terraform-modules/blob/main/azure/mssql-database/README.md

etc

**Review by Renetnielsen — DISMISSED** [review]
beautiful, small comment on documentation example.
But approved

File: azure/service-bus-topic-subscription/main.tf [hunk 4]
```diff
@@ -18,4 +18,15 @@ resource "azurerm_servicebus_subscription" "this" {
 
   max_delivery_count  = try(var.max_delivery_count, null)
   forward_to          = try(var.forward_to, null)
+}
+
+
+resource "azurerm_servicebus_subscription_rule" "sql_filter" {
+  count               = try(var.sql_filter.name != "", false) ? 1 : 0
```
Comment (djorgensendk) [inline azure/service-bus-topic-subscription/main.tf:25]: how does the var.sql_filter.name != "" act if the object is null (the default value)?

**Review by djorgensendk — APPROVED** [review]

**Review by Renetnielsen — APPROVED** [review]
LGTM +  i saw it work 👍

**Pull request merged**